sync/src/relayer/compact_block_verifier.rs

```diff
@@ -0,0 +1,94 @@
+use crate::relayer::compact_block::CompactBlock;
+use crate::relayer::error::Error;
+use ckb_protocol::{short_transaction_id, short_transaction_id_keys};
+use std::collections::HashSet;
+
+pub struct CompactBlockVerifier {
+    prefilled: PrefilledVerifier,
+    short_ids: ShortIdsVerifier,
+}
+
+impl CompactBlockVerifier {
+    pub(crate) fn new() -> Self {
+        Self {
+            prefilled: PrefilledVerifier::new(),
+            short_ids: ShortIdsVerifier::new(),
+        }
+    }
+
+    pub(crate) fn verify(&self, block: &CompactBlock) -> Result<(), Error> {
+        self.prefilled.verify(block)?;
+        self.short_ids.verify(block)?;
+        Ok(())
+    }
+}
+
+pub struct PrefilledVerifier {}
+
+impl PrefilledVerifier {
+    pub(crate) fn new() -> Self {
+        Self {}
+    }
+
+    pub(crate) fn verify(&self, block: &CompactBlock) -> Result<(), Error> {
+        let prefilled_transactions = &block.prefilled_transactions;
+        let short_ids = &block.short_ids;
+        let txs_len = prefilled_transactions.len() + short_ids.len();
+
+        // Check indices order of prefilled transactions
+        let mut prev = prefilled_transactions
```

Comment:
We can use `windows` here
```
prefilled_transactions.iter().windows(2).any(|pt| pt[0].index > pt[1].index)
```

Reply:
Thanks. Change to use `tuple_window` instead.

Reply:
`tuple_windows` will clone the iterator element if I remember correctly, and I prefer the std collection than itertools crate.

Reply:
Oh, `windows` is only implemented for `Slice`, but not `Iterator`.
Changed to use `Slice::windows` instead.

```
        let unordered = prefilled_transactions
            .as_slice()
            .windows(2)
            .any(|pt| pt[0].index >= pt[1].index);
```